### PR TITLE
FEAT: Implemented web UI for launching the text2image model.

### LINKED ISSUE #792: FEAT: UI for text2image
As we support stable diffusion model for text2image, a simple UI is needed to let users generate images on web pages. Like chat UI, we can launch a gradio page for a quick implementation.

### PR DESCRIPTION
**Pull request by hainaweiben**
Fixes #792

### CODE REVIEW

File: xinference/web/ui/src/scenes/launch_model/index.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ import LaunchCustom from './launchCustom'
 import LaunchEmbedding from './launchEmbedding'
 import LaunchLLM from './launchLLM'
 import LaunchRerank from './launchRerank'
+import LaunchText2Image from "./launchText2Image";
```
Comment (codingl2k1) [inline xinference/web/ui/src/scenes/launch_model/index.js:14]: launch model use the model type as part of the file name, e.g. `imageCard.js`, `launchImage.js`.

File: xinference/web/ui/src/scenes/launch_model/index.js [hunk 2]
```diff
@@ -70,6 +71,7 @@ const LaunchModel = () => {
             <Tab label="Embedding Models" value="2" />
             <Tab label="Rerank Models" value="3" />
             <Tab label="Custom Models" value="4" />
+            <Tab label="Text2Image MOdels" value="5" />
```
Comment (codingl2k1) [inline xinference/web/ui/src/scenes/launch_model/index.js:74]:
```suggestion
            <Tab label="Image Models" value="5" />
```

**Comment by codingl2k1** [comment]
BTW, you can run `npm run format` in `xinference/web/ui` directory to fix the lint.

File: xinference/web/ui/src/scenes/launch_model/launchText2Image.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+          return {
+            ...(await desc.json()),
+            is_builtin: registration.is_builtin,
+            is_cached: registration.cache_status,
+          };
+        }),
+      );
+
+      const builtinRerankModels = newRegistrationData.filter((v) => {
```
Comment (ChengjieLi28) [inline xinference/web/ui/src/scenes/launch_model/launchText2Image.js:60]: Naming issue. Why `rerank` happens here.

File: xinference/web/ui/src/scenes/launch_model/Text2ImageCard.js [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+      minHeight: '120px',
+    },
+    h2: {
+      margin: '10px 10px',
+      fontSize: '20px',
+    },
+    buttonsContainer: {
+      display: 'flex',
+      margin: '0 auto',
+      marginTop: '60px',
```
Comment (ChengjieLi28) [inline xinference/web/ui/src/scenes/launch_model/Text2ImageCard.js:128]: Remove these two lines to keep align with the latest main code.

File: xinference/web/ui/src/scenes/launch_model/Text2ImageCard.js [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+              })()}
+          </Stack>
+        </div>
+        {hover ? (
+          <p style={styles.instructionText}>
+            Click with mouse to launch the model
+          </p>
+        ) : (
+          <p style={styles.instructionText}></p>
+        )}
```
Comment (ChengjieLi28) [inline xinference/web/ui/src/scenes/launch_model/Text2ImageCard.js:265]: Change to this:
```
{!selected && hover && (
          <p style={styles.instructionText}>
            Click with mouse to launch the model
          </p>
        )}
```
May be you need to rebase to the latest main code. You are using the old version code here.

**Comment by ChengjieLi28** [comment]
Please update some screenshots to show the effect of your PR.

File: xinference/web/ui/src/scenes/launch_model/Text2ImageCard.js (hunk 5 above)
Comment (hainaweiben) [reply xinference/web/ui/src/scenes/launch_model/Text2ImageCard.js:265]: Alright, I'll make the change.

**Comment by hainaweiben** [comment]
Sure

webUI_imagelaunch

**Comment by codingl2k1** [comment]
Use `imageCard.js` as the file name istead of `ImageCard.js`.

**Comment by codingl2k1** [comment]
It's better to move the tab `IMAGE MODELS` to the front of tab `CUSTOM MODELS`. We keep `CUSTOM MODELS` the last tab.

**Comment by codingl2k1** [comment]
Can you keep the tab order consistent between launch model and running model pages?
3ed6d382-f00a-4d34-ad48-a285c1e8d186
daf78f65-4421-495d-af20-d5e209ee3caf

**Comment by hainaweiben** [comment]
Is the order according to the sequence of the launch model or the running model?

**Comment by codingl2k1** [comment]
We can keep the tab order in the launch model page.

**Review by codingl2k1 — APPROVED** [review]

**Comment by codingl2k1** [comment]
We can add gpu and replica options in the future (in other PR).

**Pull request merged**